mdal/frmts/mdal_hec2d.cpp

```diff
@@ -431,7 +449,7 @@ std::shared_ptr<MDAL::MemoryDataset2D> MDAL::DriverHec2D::readElemOutput( const
             {
               assert( bed_elevation );
               double bed_elev = bed_elevation->scalarValue( eInx );
-              if ( std::isnan( bed_elev ) || fabs( bed_elev - val ) > eps ) // change from bed elevation
+              if ( std::isnan( bed_elev ) || fabs( val - bed_elev ) > ( val + bed_elev )*eps )  // change from bed elevation
```

Comment:
not sure if I understand this comparison

Reply:
epsilon is the difference between 1.0 and the next value representable by the floating-point

For a float epsilon # 0.000000X 
To compare float (7 signifiant digits) that are not next to 1.0, need to transform epsilon to give it the order of magnitde of the compared float. 
Several way can be used depending on context, here the sum is enought.